resources/fetching/factory/aws_factory.go

```diff
@@ -39,6 +42,59 @@ import (
 	"github.com/elastic/cloudbeat/resources/providers/awslib/sns"
 )
 
+type AwsAccount struct {
+	*awslib.Identity
+	aws.Config
+}
+
+type wrapResource struct {
+	wrapped  fetching.Resource
+	identity *awslib.Identity
+}
+
+func (w *wrapResource) GetMetadata() (fetching.ResourceMetadata, error) {
+	mdata, err := w.wrapped.GetMetadata()
+	if err != nil {
+		return mdata, err
+	}
+	mdata.AwsAccountAlias = w.identity.Alias
+	mdata.AwsAccountId = w.identity.Account
+	return mdata, nil
+}
+
+func (w *wrapResource) GetData() any              { return w.wrapped.GetData() }
+func (w *wrapResource) GetElasticCommonData() any { return w.wrapped.GetElasticCommonData() }
+
+func NewCisAwsOrganizationFactory(log *logp.Logger, rootCh chan fetching.ResourceInfo, accounts []AwsAccount) FetchersMap {
+	m := make(FetchersMap)
+	for _, account := range accounts {
+		ch := make(chan fetching.ResourceInfo)
+		go func(identity *awslib.Identity) {
+			for resourceInfo := range ch {
+				rootCh <- fetching.ResourceInfo{
+					Resource: &wrapResource{
+						wrapped:  resourceInfo.Resource,
+						identity: identity,
+					},
+					CycleMetadata: resourceInfo.CycleMetadata,
+				}
+			}
+		}(account.Identity)
```

Comment:
This is unfortunately necessary to correctly tag resources with the AWS account id / alias.

The benchmark refactoring might make it easier to properly use the correct account in the data provider but it's still not possible without some extra refactoring.

Thus, I propose to push this "small hack" which will make the feature work functionally and later figure out a cleaner way to implement it.